openpgp/updater.go

```diff
@@ -0,0 +1,145 @@
+/*
+   Hockeypuck - OpenPGP key server
+   Copyright (C) 2012-2014  Casey Marshall
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Affero General Public License as published by
+   the Free Software Foundation, version 3.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Affero General Public License for more details.
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package openpgp
+
+import (
+	"github.com/jmoiron/sqlx"
+)
+
+type updater interface {
+	UpdatePubkey(p *Pubkey) error
+	UpdateSubkey(s *Subkey) error
+	UpdateUserId(u *UserId) error
+	UpdateUserAttribute(u *UserAttribute) error
+	UpdateSignature(s *Signature) error
+	UpdatePubkeyRevsig(p *Pubkey, s *Signature) error
+	UpdateSubkeyRevsig(sk *Subkey, s *Signature) error
+	UpdateUidRevsig(u *UserId, s *Signature) error
+	UpdateUatRevsig(u *UserAttribute, s *Signature) error
+	UpdatePrimaryUid(p *Pubkey, u *UserId) error
+	UpdatePrimaryUat(p *Pubkey, u *UserAttribute) error
+}
+
+type postgresUpdater struct {
+  *sqlx.Tx
+}
+
+func newPostgresUpdater(tx *sqlx.Tx) updater {
```

Comment:
That's a good question. In this case it may make sense to call this the newPostgresUpdater as I imagine the neo4jUpdater would likely not be able to accept `sqlx.Tx`